tests/tests.rs

```diff
@@ -189,14 +189,86 @@ fn eq_option_err_07() {
                "failed to parse 'a' cause invalid digit found in string");
 }
 
-#[cfg(not(feature = "eq-separator"))]
+#[cfg(not(any(feature = "eq-separator", feature = "short-space-opt")))]
 #[test]
 fn no_eq_separator_01() {
     let mut args = Arguments::from_vec(to_vec(&["-w=a"]));
     let value: Result<Option<u32>, Error> = args.opt_value_from_str("-w");
     assert_eq!(value.unwrap(), None);
 }
 
+#[cfg(feature = "short-space-opt")]
+#[test]
+fn space_option_01() {
+    let mut args = Arguments::from_vec(to_vec(&["-w10"]));
+    let value: Option<u32> = args.opt_value_from_str("-w").unwrap();
+    assert_eq!(value.unwrap(), 10);
+}
+
+#[cfg(feature = "short-space-opt")]
+#[test]
+fn space_option_02() {
+    let mut args = Arguments::from_vec(to_vec(&["-w--width"]));
+    let value: Option<String> = args.opt_value_from_str(["-w", "--width"]).unwrap();
+    assert_eq!(value.unwrap(), "--width");
+}
+
+#[cfg(feature = "short-space-opt")]
+#[test]
+fn space_option_03() {
+    let mut args = Arguments::from_vec(to_vec(&["-w'10'"]));
+    let value: Option<u32> = args.opt_value_from_str(["-w", "--width"]).unwrap();
+    assert_eq!(value.unwrap(), 10);
+}
+
+#[cfg(feature = "short-space-opt")]
+#[test]
+fn space_option_04() {
+    let mut args = Arguments::from_vec(to_vec(&["-w\"10\""]));
+    let value: Option<u32> = args.opt_value_from_str(["-w", "--width"]).unwrap();
+    assert_eq!(value.unwrap(), 10);
+}
+
+#[cfg(all(feature = "short-space-opt", not(feature = "eq-separator")))]
+#[test]
+fn space_not_eq_option_01() {
+    let mut args = Arguments::from_vec(to_vec(&["-w=10"]));
+    let value: Option<String> = args.opt_value_from_str("-w").unwrap();
+    assert_eq!(value.unwrap(), "=10");
+}
```

Comment:
This is very bad. Cargo features are additive, so we cannot change the behavior that much and must assume that a user have enabled all the flags.

I think we should return an error, when `eq-separator` is disabled, but `=` is present. Otherwise, it can silently return `10` instead of `=10`.

Reply:
Looks like I was not clear. You should have used the `OptionWithoutAValue` error. Adding a new error is a breaking change and I don't want that.

Reply:
I added an error for the edge case when `short-space-opt` is enabled but `eq-separator` is not and the user gives `-w=10`. If the user really wants to give '-10' then it can be quoted. The tests reflect this behavior.

Reply:
My bad, I used `OptionWithoutAValue` in the latest commit.